File: scripts/remove_bg.py
import sys
import os
from PIL import Image

try:
    from rembg import remove
except ImportError:
    print("rembg is not installed yet. Please wait for the installation to finish.")
    sys.exit(1)

def main():
    input_path = "public/Profile2.jpeg"
    output_path = "public/Profile3.png"

    if not os.path.exists(input_path):
        print(f"Error: {input_path} not found.")
        sys.exit(1)

    print(f"Removing background from {input_path}...")
    try:
        input_image = Image.open(input_path)
        output_image = remove(input_image)
        output_image.save(output_path)
        print(f"Success! Background-removed image saved to {output_path}")
    except Exception as e:
        print(f"An error occurred: {e}")
        sys.exit(1)

if __name__ == "__main__":
    main()

File: scripts/remove_bg.swift
import Foundation
import Vision
import CoreImage
import CoreImage.CIFilterBuiltins
import AppKit

func removeBackground(inputPath: String, outputPath: String) {
    let inputUrl = URL(fileURLWithPath: inputPath)
    let outputUrl = URL(fileURLWithPath: outputPath)
    
    guard let ciImage = CIImage(contentsOf: inputUrl) else {
        print("Error: Could not load image from \(inputPath)")
        exit(1)
    }
    
    let request = VNGenerateForegroundInstanceMaskRequest()
    let handler = VNImageRequestHandler(ciImage: ciImage, options: [:])
    
    do {
        print("Running Vision background removal...")
        try handler.perform([request])
        guard let result = request.results?.first else {
            print("Error: Vision request returned no results.")
            exit(1)
        }
        
        let maskBuffer = try result.generateScaledMaskForImage(
            forInstances: result.allInstances,
            from: handler
        )
        let maskImage = CIImage(cvPixelBuffer: maskBuffer)
        
        let filter = CIFilter.blendWithMask()
        filter.inputImage = ciImage
        filter.maskImage = maskImage
        filter.backgroundImage = CIImage.empty()
        
        guard let outputCIImage = filter.outputImage else {
            print("Error: Filter output is empty.")
            exit(1)
        }
        
        let context = CIContext()
        guard let cgImage = context.createCGImage(outputCIImage, from: outputCIImage.extent) else {
            print("Error: Could not create CGImage.")
            exit(1)
        }
        
        let size = NSSize(width: cgImage.width, height: cgImage.height)
        let rep = NSBitmapImageRep(cgImage: cgImage)
        rep.size = size
        
        guard let pngData = rep.representation(using: .png, properties: [:]) else {
            print("Error: Could not represent as PNG.")
            exit(1)
        }
        
        try pngData.write(to: outputUrl)
        print("Success! Background-removed image saved to \(outputPath)")
    } catch {
        print("Error processing background removal: \(error)")
        exit(1)
    }
}

let arguments = CommandLine.arguments
if arguments.count < 3 {
    print("Usage: swift remove_bg.swift <input_path> <output_path>")
    exit(1)
}

removeBackground(inputPath: arguments[1], outputPath: arguments[2])
